### PR TITLE
Interrupt correctly

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: migrator/src/main/java/com/redhat/lightblue/migrator/ConsistencyCheckerController.java [hunk 1]
```diff
@@ -206,109 +206,103 @@ private boolean migrationJobsExist() {
...
-                        LOGGER.info("There are migration jobs for {}, not running consistency checker this time",
+        LOGGER.debug("Starting consistency checker controller for {} with period {}",migrationConfiguration.getConfigurationName(),
+                     migrationConfiguration.getPeriod());
+        long period=parsePeriod(migrationConfiguration.getPeriod());
+        while(!stopped) {
+            Breakpoint.checkpoint("CCC:start");
+            LOGGER.debug("Consistency checker {} woke up",migrationConfiguration.getConfigurationName());
+            // Lets update our configuration first
+            
+            MigrationConfiguration newCfg=migrationConfiguration;
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/ConsistencyCheckerController.java]: Does this need to be set with each iteration?

File: migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java [hunk 2]
```diff
@@ -27,16 +27,17 @@
...
-    public AbstractController(Controller controller, MigrationConfiguration migrationConfiguration, String threadGroupName) {
-        this.migrationConfiguration = migrationConfiguration;
-        this.controller = controller;
-        lbClient = controller.getLightblueClient();
-        locking = lbClient.getLocking("migration");
-        if (migrationConfiguration.getMigratorClass() == null) {
-            migratorClass = DefaultMigrator.class;
-        } else {
+    protected final HashSet<String> myLocks=new HashSet<>();
+    protected boolean stopped=false;
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java]: Is stopped read in this class? If not, does it maybe belong on a lower level?

File: migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java [hunk 3]
```diff
@@ -47,6 +48,11 @@ public AbstractController(Controller controller, MigrationConfiguration migratio
         migratorThreads = new ThreadGroup(threadGroupName);
     }
 
+    public void setStopped() {
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java]: just stop()?

File: migrator/src/main/java/com/redhat/lightblue/migrator/ConsistencyCheckerController.java (hunk 1 above)
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/ConsistencyCheckerController.java]: It runs once every period, and reloads the configuration to see if it should still run.

File: migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java (hunk 2 above)
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java]: It is not read here, but when the thread needs to stop, it is set at this level. All controller classes inherit AbstractController.

File: migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java (hunk 3 above)
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/AbstractController.java]: Clashes with Thread.stop()

**Pull request merged**